weather.go

```diff
@@ -1,5 +1,138 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type apiResponse struct {
+	Name    string
+	Weather []struct {
+		Main string
+	}
+	Main struct {
+		Temp float64
+	}
+}
+
+type apiErrorResponse struct {
+	Cod     string
+	Message string
+}
+
+type Conditions struct {
+	City               string
+	OneWord            string
+	TemperatureCelsius float64
+}
+
+func RunCLI(args []string) (output string) {
+
+	token := os.Getenv("WEATHERAPP_TOKEN")
+	if len(token) == 0 {
+		fmt.Fprintf(os.Stderr, "please set env variable, WEATHERAPP_TOKEN \n")
+		os.Exit(2)
+	}
+
+	request, err := Request(args, token)
+	if err != nil {
+		log.Printf("an error has occured, %v", err)
+		os.Exit(2)
+	}
+
+	response, err := Response(request)
+	if err != nil {
+		log.Printf("an error has occured, %v", err)
+		os.Exit(2)
+	}
+
+	conditions, err := ParseResponse(response)
+	if err != nil {
+		log.Printf("problem parsing API response', %v", err)
+		os.Exit(2)
+	}
+
+	output = fmt.Sprintf("City: %s\nWeather: %s\nCelsius: %v\n", conditions.City, conditions.OneWord, conditions.TemperatureCelsius)
```

Comment:
If you think about this program as a library, which is something I always encourage, what will users generally want to do? They will want to get the weather conditions, so it would make sense for there to be some function like `GetConditions` that returns a `Conditions` struct. It's then trivial to print that (you can define a `String` method on `Conditions`, for example), but it's not trivial to extract the _data_ from the printed version.